jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -6602,6 +6602,13 @@ private static char convEndian(boolean isBigEndian, char value) {
 	 */
 	public native boolean isFlattened(Field field);
 
+	/**
+	 * Determines the size of an object in bytes
```

Comment:
does ojdk include header size?

is primitive `class Foo { int bar }` 8bytes?

Reply:
Yes it does appear to include headers, although the actual size is 16 bytes (which checks out given that `unsafe.valueHeaderSize(Foo.class)` is 12)